Fluid/Filters/MiscFilters.cs

```diff
@@ -614,14 +614,23 @@ private static async ValueTask WriteJson(Utf8JsonWriter writer, FluidValue input
                                 value = access.Get(obj, name, ctx);
                             }
 
+                            if (value is object)
+                            {
+                                var occurrences = stack.GroupBy(o => o).ToDictionary(o => o.Key, o => o.Count());
+                                if (occurrences.ContainsKey(value) && occurrences[value] > 1)
```

Comment:
Wouldn't this be simpler by using just `HashSet<object>` and if `stack.Add(value)` returns false it's a cycle? Initial call site just needs to pass a `new HashSet<object>()` or check if null and initialize.

Would perform a lot better unless I'm missing something.

Reply:
I'll update the logic shortly.

Reply:
One issue with the `HashSet<>` approach is that once you initialize it, the same reference is being used when traversing all of the edges of the object graph.  This means that if two adjacent properties have references to the same object, a cycle will be erroneously be detected.  To solve that, a new instance needs to be created as we recurse to another call of `WriteJson`.

Reply:
Updated to use a `HashSet<>`.